Define Step Uploads action before talking about it (#3126)
buildkite/docs: pages/pipelines/configure/defining_steps.md

If your pipeline has more than one step and you have multiple agents available to run them, they will automatically run at the same time. If your steps rely on running in sequence, you can separate them with [wait steps](/docs/pipelines/configure/step-types/wait-step). This will ensure that any steps before the 'wait' are completed before steps after the 'wait' are run.

> 📘

> > It is also possible to upload additional steps while a build is in progress. This isn't a step type, instead it is an action performed by an agent or user. See [Dynamic pipelines](/docs/pipelines/configure/dynamic_pipelines.md) for more information.

Reviewer: Step uploads are always done by an agent, not by a user

```suggestion
> It is also possible to upload additional steps while a build is in progress. This isn't a step type, instead it is an action performed by an agent within the context of a job. See [Dynamic pipelines](/docs/pipelines/configure/dynamic_pipelines.md) for more information.
```

Author: Oh that's interesting.  Is it "within" or "with"? Like does it have to be an agent currently running some step or is it any agent running anywhere that is observing the state of a job.

Reviewer: oops - yeah, should be "within". Has to be done using a job token, which is only issued when an agent accepts a job, and is tied to the lifetime of that job
